service/controller/v2/resource/endpoints/current.go

```diff
@@ -35,6 +37,9 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		manifest, err := r.k8sClient.CoreV1().Endpoints(namespace).Get(masterEndpointsName, apismetav1.GetOptions{})
 		if apierrors.IsNotFound(err) {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the master endpoints in the Kubernetes API")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
```

Comment:
If we cancel the resource when endpoints are not created, then how can we ever create them?

Reply:
I have to check on this.

Reply:
Yes this was wrong. I fixed it like for the virtual networks. Now we check if the interfaces are there and if not we cancel the resource.